### PR TITLE
fix: null check in ListenerCollection notify method

### PR DESCRIPTION
**Pull request by kkocel**
Fix for the issue I've encountered in the wild:

```
Caused by: java.lang.NullPointerException: Cannot invoke "java.util.function.Consumer.accept(Object)" because "listener" is null
    at com.microsoft.playwright.impl.ListenerCollection.notify(ListenerCollection.java:44)
    at com.microsoft.playwright.impl.PageImpl.didClose(PageImpl.java:234)
    at com.microsoft.playwright.impl.PageImpl.handleEvent(PageImpl.java:222)
    at com.microsoft.playwright.impl.Connection.dispatch(Connection.java:292)
    at com.microsoft.playwright.impl.Connection.processOneMessage(Connection.java:211)
    at com.microsoft.playwright.impl.ChannelOwner.runUntil(ChannelOwner.java:118)
    at com.microsoft.playwright.impl.Connection.sendMessage(Connection.java:130)
    at com.microsoft.playwright.impl.ChannelOwner.sendMessage(ChannelOwner.java:106)
    at com.microsoft.playwright.impl.FrameImpl.fillImpl(FrameImpl.java:335)
    at com.microsoft.playwright.impl.FrameImpl.lambda$fill$14(FrameImpl.java:325)
    at com.microsoft.playwright.impl.LoggingSupport.lambda$withLogging$0(LoggingSupport.java:36)
    at com.microsoft.playwright.impl.LoggingSupport.withLogging(LoggingSupport.java:47)
    at com.microsoft.playwright.impl.ChannelOwner.withLogging(ChannelOwner.java:89)
    at com.microsoft.playwright.impl.LoggingSupport.withLogging(LoggingSupport.java:35)
    at com.microsoft.playwright.impl.FrameImpl.fill(FrameImpl.java:325)
    at com.microsoft.playwright.impl.LocatorImpl.fill(LocatorImpl.java:235)
    at com.microsoft.playwright.Locator.fill(Locator.java:2988)
```

### CODE REVIEW

**Comment by kkocel** [comment]
@microsoft-github-policy-service agree [company="{your company}"]

File: playwright/src/main/java/com/microsoft/playwright/impl/ListenerCollection.java [hunk 1]
```diff
@@ -41,7 +41,9 @@ <T> void notify(EventType eventType, T param) {
     }
 
     for (Consumer<?> listener: new ArrayList<>(list)) {
-      ((Consumer<T>) listener).accept(param);
+      if (listener != null) {
```
Comment (yury-s) [inline playwright/src/main/java/com/microsoft/playwright/impl/ListenerCollection.java:44]: Having `null` listeners in the list is an error, we should instead throw when the client attempts to add `null` as the listener.

File: playwright/src/test/java/com/microsoft/playwright/impl/ListenerCollectionTest.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package com.microsoft.playwright.impl;
+
+import org.junit.jupiter.api.Test;
+
+class ListenerCollectionTest {
+
+  @Test
+  void shouldHandleNullListener() {
```
Comment (yury-s) [inline playwright/src/test/java/com/microsoft/playwright/impl/ListenerCollectionTest.java:8]: We prefer integration tests in the project, can you convert it in a test that uses Playwright public API?

**Review by yury-s — COMMENTED** [review]
Is this a fix for https://github.com/microsoft/playwright-java/issues/1659 ?

**Comment by kkocel** [comment]
> Is this a fix for #1659 ?

nope, it's another bug I've encountered

File: playwright/src/test/java/com/microsoft/playwright/impl/ListenerCollectionTest.java (hunk 2 above)
Comment (kkocel) [reply playwright/src/test/java/com/microsoft/playwright/impl/ListenerCollectionTest.java]: sure, changed

File: playwright/src/main/java/com/microsoft/playwright/impl/ListenerCollection.java (hunk 1 above)
Comment (kkocel) [reply playwright/src/main/java/com/microsoft/playwright/impl/ListenerCollection.java]: changed

**Review by yury-s — APPROVED** [review]

**Comment by kkocel** [comment]
@yury-s I've seen that some tests randomly crashed and I don't have permissions to restart them.

**Comment by yury-s** [comment]
> @yury-s I've seen that some tests randomly crashed and I don't have permissions to restart them.

These are old flakes, your PR is fine.

**Pull request merged**